lib/api/controllers/security.js

```diff
@@ -1480,6 +1333,176 @@ class SecurityController extends NativeController {
     });
     return securityDocument;
   }
+
+  /**
+   * @param {Request} request
+   * @param {{ _id: string, _kuzzle_info: object }} pojoUser
+   * @returns {Promise<unknown>}
+   * @private
+   */
+  async _persistUser(request, pojoUser) {
+    let loadedUser;
+
+    // Add metadata
+    pojoUser._kuzzle_info = {
+      author: request.context.user ? String(request.context.user._id) : null,
+      createdAt: Date.now(),
+      updatedAt: null,
+      updater: null
+    };
+
+    try {
+      loadedUser = await this.kuzzle.repositories.user.load(pojoUser._id);
+    }
+    catch (error) {
+      if (!(error instanceof NotFoundError)) {
```

Comment:
It's better to use error codes now
```suggestion
      if (error.id !== 'services.storage.not_found') {
```